pkg/commands/options/publish.go

```diff
@@ -73,6 +78,9 @@ func AddPublishArg(cmd *cobra.Command, po *PublishOptions) {
 	if dockerRepo, exists := os.LookupEnv("KO_DOCKER_REPO"); exists {
 		po.DockerRepo = dockerRepo
 	}
+	if dockerRepoHost, exists := os.LookupEnv("KO_DOCKER_REPO_HOST"); exists {
```

Comment:
`KO_DOCKER_REPO_HOST` doesn't feel like the right name for this, but I don't think I understand this well enough to suggest a better name.

Reply:
Totally open to other possibilities. I even thought about using some weird syntax for `KO_DOCKER_REPO`, but that seemed more dangerous.

Reply:
Is this like `curl --resolve`? If so, how about something with `RESOLVE` in the name?

And since the `DialContext` argument allows for host+port, how about `ADDRESS` instead of `HOST`?

E.g., `KO_REPO_RESOLVE_ADDRESS`

Reply:
@halvards yeah indeed, it's like `--resolve`